R/Renderer.R

```diff
@@ -33,10 +33,36 @@ Renderer <- R6::R6Class( # nolint: object_name_linter.
       if (missing(yaml_header)) {
         yaml_header <- md_header(yaml::as.yaml(list(title = "Report")))
       }
-      parsed_yaml <- yaml_header
+
+      private$report_type <- get_yaml_field(yaml_header, "output")
+      format_code_block_for_slide <- function() {
+        paste0(
+          c(
+            "code_block <- function (code_text) {",
+            "  df <- data.frame(code_text)",
+            "  ft <- flextable::flextable(df)",
+            "  ft <- flextable::delete_part(ft, part = 'header')",
+            "  ft <- flextable::autofit(ft, add_h = 0)",
+            "  ft <- flextable::fontsize(ft, size = 7, part = 'body')",
+            "  ft <- flextable::bg(x = ft, bg = 'lightgrey')",
+            "  ft <- flextable::border_outer(ft)",
+            "  if (flextable::flextable_dim(ft)$widths > 8) {",
+            "    ft <- flextable::width(ft, width = 8)",
+            "  }",
+            "  ft",
+            "}"
+          ),
+          collapse = "\n"
+        )
+      }
       parsed_global_knitr <- sprintf(
-        "\n```{r setup, include=FALSE}\nknitr::opts_chunk$set(%s)\n```\n",
-        capture.output(dput(global_knitr))
+        "\n```{r setup, include=FALSE}\nknitr::opts_chunk$set(%s)\n%s\n```\n",
+        capture.output(dput(global_knitr)),
+        if (identical(private$report_type, "powerpoint_presentation")) {
+          format_code_block_for_slide()
+        } else {
+          character(0)
+        }
```

Comment:
Well, if you keep it here, it doesn't have to be a function, a string is sufficient.

But bear in mind
```
> sprintf("%s hello", character(0))
character(0)
```